libs/utils/include/celix_properties.h

```diff
@@ -385,14 +512,264 @@ celix_properties_getVersion(const celix_properties_t* properties, const char* ke
+
+/**
+ * @brief Get a property value as an array of celix_version_t entries, making a copy of the array.
+ *
+ *
+ * This function retrieves the value of a property, interpreting it as an array of celix_version_t* entries. If the
+ * underlying type of the property value is a celix_version_t* array, a copy of the array is returned. If the underlying
+ * type is a string, the string is converted to an array of celix_version_t* if possible. If the property is not set,
+ * its value is not an array of celix_version_t* entries or its value cannot be converted to a celix_version_t* array,
+ * the default value is returned as a copy.
+ *
+ * The returned array list is configured with a remove callback so that the destruction of the array list will also
+ * free the celix_version_t entries in the array list.
+ *
+ * @param[in] properties The property set to search.
+ * @param[in] key The key of the property to get.
+ * @param[in] defaultValue The default value to return if the property is not set or its value is not an array of
+ * celix_version_t entries.
+ * @param[out] list A copy of the found list, a new array list with celix_version_t values or a copy of the default
+ * value if the property is not set, its value is not an array of celix_version_t entries or its value cannot be
+ * converted to an array of celix_version_t entries.
+ * @return CELIX_SUCCESS if the operation was successful, CELIX_ENOMEM if there was not enough memory to create the
+ *        array list. Note if the key is not found, the return status is still CELIX_SUCCESS.
+ */
+CELIX_UTILS_EXPORT celix_status_t celix_properties_getAsVersionArrayList(const celix_properties_t* properties,
+                                                                         const char* key,
+                                                                         const celix_array_list_t* defaultValue,
+                                                                         celix_array_list_t** list);
+
+/**
+ * @brief Get the property value as an array of celix_version_t entries without copying.
+ *
+ * This function provides a non-owning, read-only access to a property value interpreted as an array of celix_version_t
+ * entries. It returns a const pointer to the array. If the property is not set or its value is not an array of
+ * celix_version_t entries, the default value is returned.
+ *
+ * @param[in] properties The property set to search.
+ * @param[in] key The key of the property to get.
+ * @param[in] defaultValue The value to return if the property is not set or its value is not an array of
+ * celix_version_t entries.
+ * @return A const pointer to the property value interpreted as an array of celix_version_t entries, or the default
+ * value if the property is not set or its value is not an array of celix_version_t entries. The returned pointer should
+ * not be modified or freed.
+ */
+CELIX_UTILS_EXPORT const celix_array_list_t* celix_properties_getVersionArrayList(
```

Comment:
A single `celix_properties_getArrayList` would be better?

Reply:
I think that would be more error prone. Because this way the user known the returned array list is a array list with entry type version. 

With exception of generic printing / serialization I do not see a use case to retrieve all array list types. And for that use case we already have `celix_properties_getEntry`